### PR TITLE
Fix equality comparison with a record with a with expression

### LINKED ISSUE #2507: Equality comparison with a record with a `with` expression results in broken code
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAbRAXMcC2ADgIYBOiAKgBYCWAdgOYDOYAFAGbUmNZpV30BKVo0Rxa8XjQZCAvAB0Sc2hAgcuOGWGBgRY%2BGADu1LJQVKVFyxYBisWGE1ruAOluwzyq14gAhUg7AnLGc-RUVPbwgAX0gAGnAoESwsfkZUMABtBIhQK0gAEiwATwJkNEg6LDjslQKANyJ0AFcyzJqLAAZ470gASXFEWiwAZWoAL2RurwAWdrAAXRqoqZy5guLS9Iqh6rz8hubWrO8ARhXLSABZIgAPABk6RDvB%2BhNdrxOAJg65xYtlmq5C6FEqtSCDeAAeTYD1oiGGxUw7ws9UaLXSxy8n3OKIAouJobDJnMIJB0GxIL8lucgSiQZtygAjOzoZG1fZoo4ksAAZhxtX6IhIWGsdEaADlEAZ0I82RYsCQWlT-jS1vSwczYKz%2BajDhjudN%2BaThsRoIgfIg2LAyAAFUhEPDYRAkOUqBVKqx-FQAiy09kbDUs1269FoTFWACsRqgJqIZotVrId1gBmd0CIIn6dVg6ZSsForog7sQyu9qr2Aa2mu1aoOoba3gAbNHILH45brYgAKoEUokdOZ2jZ3PUfOFwKNESl6Ll4GVplBnUcvVh7kAdhbbfNHbIAGF0BnGLv89xFdAsNbx2xJyXPdTAWr51Bq8Hl-Xw5YAByb03bxOIS5EDwRlnSvG9pzAH0VD9Ul1SrRda05fVvAATh-OM-07Y90DHaNr3QKc7xVB8K1BeCtVfOsuVOLpuVbX8AEE2CwZ1jzwPAiHHYsIKg1ZSIZZ8EL2KjkI%2BM46K3BNO2GIDqDgHCCzw8CiLLEi5zIhcKKXETV1ObEJMY5jnRkvA5K1XDuW4lSZzUukn0gF9tKQ3SPj5AyMIYkhYCacQABFEBlUyWJdaMrMsL0bN9R8NKgSpKOchsPkNOjrhuXo2CoQZhkoa0sAAdWoeA3mjH5rMg2c7Ji7YqicldEqsE4oxS250sy2hcQI%2BEcuFAqisocdG1K8L7yi-iwTi2r325E5m2atLaA4G5IT7IgLxIXEbgIMhGEYUdFO5T8hosCLyts-0qomxC6o-CwTg3OaACVRGteBeuKg0jpUE7eK0aKBOq%2BLrum79Huekh4HFJpgOdaFehYvA0mjE4eIq87-rwJp0BSGU4VsEgOOSZ1yDIybqI%2BNC6KeuBwcuTHscePGCeC8cYEoe1z2dAB9Iw%2BspMqfpgqA4PKS7hISm6VG%2BFtUs8kgiCKSESAebg3v66NDpRs7YPs0XgR0%2BrLE%2BcSehlkg5YVpXqG4SHoZIWH4cR6bNdG9T0bp6gccQRnVuC4nNlJ0SrE%2BfSell%2BXFeVrBaaxj2GetJnQJbaA2bljmSG5wq3md6C-vGnYA5coO3JN24ADVOR8Oh4H4VXxw1-nUe1i786uqbvE%2BZKS5uawfPPPbK-EGvM7Vj7s741285q1uyaDpqu982AsAAcWwDatsQHa9tr9XPogb7G6F%2BzHOnwPDdmk33c9nwcOgABrHxU9v7BGEhWhhgdRBsKh-bvDC46RpzmNciNYxZA3bvdHokppSPAtFgVMgw-aIH8hwWgxg9oMXEIBW2jtf6KlvMNYiLtKr-WPqAtuWIQahxlPQWgPdaB93zKMGhq0mhkHILAfoSAhirT2mBTqY9fpAM0iAvW4tuSfAplQ4KtAeF1CePmegWCQIkGQXQNBJ4%2BGEQIapIhaNAxaRPoXSwPJaIXxjkSO4DpGTwCIPuWAu0GBQM9po-B-9CGAInsAwG5CrA8mNl4SAa9nTUEdNw9AABpRAiACCcMGFgfoD8iD0JHt4fCWi3E6I8cQ-RIi6T6wlhAHkIcAnXySbfIkjBPLeXEJKbgiB4DR3pnCNe21doaNCnggRgt1hVVIaIsBXgeTFxKaQKSZBfJW2gCQEJYoWLwC7Kg-M-loAHjlnmH%2BXg0muK%2BgA8e2SvEFwNhYHkncAlBOmaErAjQEReSaKeJoBAERFCREpfhDctaH2blPMhM9jFzwCZE6JqUbbKOhKU2g5THg4I%2BB0XeCxdmCM8cI7xvzjnnwCTcuSUcEDElScpbR0QEjzCmBURg1hdrpC2SAKIQA)

#### Code

```fsharp
let compareThings (first: Thing) (second: Thing) =
    first = { second with
                Foo = first.Foo
                Bar = first.Bar
            }
```
            

#### Result

```fsharp
let compareThings (first: Thing) (second: Thing) =
    first = { second with
        Foo = first.Foo
        Bar = first.Bar
    }
```
            
#### Problem description

The record properties don't get indented properly. This seems to happen in default, same-column, and Stroustrup alignment, so it seems fully independent of those settings.

#### Extra information

- [x] The formatted result breaks my code.
- [x] The formatted result gives compiler warnings.
- [x] I or my company would be willing to help fix this.

#### Options

Fantomas master branch at 2022-09-16T18:11:37Z - 5182588ea61183f658c572eb55f3f9cc926c263a

```fsharp
    { config with
                MultilineBlockBracketsOnSameColumn = true }
```

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#ignore-files-fantomasignore)?</sub>

### PR DESCRIPTION
**Pull request by josh-degraw**
Closes #2507.

### CODE REVIEW

File: src/Fantomas.Core.Tests/RecordTests.fs [hunk 1]
```diff
@@ -2076,3 +2076,48 @@ and seekReadParamExtras (retRes, paramsRes) (idx: int) =
                 Attributes = enum<ParameterAttributes> flags
                 CustomAttrs = cas }
 """
+
+[<Test>]
+let ``Equality comparison with a `with` expression should format correctly with default alignment, 2507`` () =
```
Comment (dawedawe) [inline src/Fantomas.Core.Tests/RecordTests.fs:2081]: Please start tests with [lowercase](https://fsprojects.github.io/fantomas/docs/contributors/Pull%20request%20ground%20rules.html)

**Review by dawedawe — COMMENTED** [review]
Thanks!
Please add an entry to the CHANGELOG.md

File: src/Fantomas.Core.Tests/RecordTests.fs (hunk 1 above)
Comment (josh-degraw) [reply src/Fantomas.Core.Tests/RecordTests.fs]: Done

File: src/Fantomas.Core.Tests/RecordTests.fs [hunk 2]
```diff
@@ -2076,3 +2076,48 @@ and seekReadParamExtras (retRes, paramsRes) (idx: int) =
...
+    |> should
+        equal
+        """
+let compareThings (first: Thing) (second: Thing) =
+    first = { second with
+                Foo = first.Foo
+                Bar = first.Bar }
+"""
+
+[<Test>]
```
Comment (nojaf) [inline src/Fantomas.Core.Tests/RecordTests.fs:2102]: Not the end of the world, but I would typically not add an extra test for `{ config with MultilineBlockBracketsOnSameColumn = true }`. 
The problem was present with the default configuration and should have been reported as such.
There is no more code coverage by this additional test.

Just a heads up for the future.
What would be a more interesting test case would be switching the order of the arguments in the infix application.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 ### Fixed
 * Items.Item \<index> <- value gives wrong code. [#2498](https://github.com/fsprojects/fantomas/issues/2498)
+*  Fix equality comparison with a record with a with expression. [2513](https://github.com/fsprojects/fantomas/pull/2513)
```
Comment (nojaf) [inline CHANGELOG.md]: Nitpick: please try and use the exact structure to add a changelog entry.

**Review by nojaf — APPROVED** [review]
Many thanks!

**Pull request merged**